services/rss/controllers/feed.ts

```diff
@@ -0,0 +1,199 @@
+import db from "@library/db";
+const log = require("loglevel");
+const { Podcast } = require("podcast");
+const { v5, validate } = require("uuid");
+const {
+  feedPath,
+  receivingPublicKey,
+  podcastNamespace,
+  valueRecipient,
+  valueTimeSplit,
+} = require("../library/rssUtils");
+const {
+  buildAlbumFeed,
+  buildArtistFeed,
+  buildPodcastFeed,
+} = require("../library/feedBuilder");
+
+// Error handling
+// Ref: https://stackoverflow.com/questions/43356705/node-js-express-error-handling-middleware-with-router
+const handleErrorAsync = (fn) => async (req, res, next) => {
+  try {
+    await fn(req, res, next);
+  } catch (error) {
+    next(error);
+  }
+};
+
+exports.getArtistFeed = handleErrorAsync(async (req, res, next) => {
+  const { feedId } = req.params;
+
+  const validUuid = validate(feedId);
+
+  if (!validUuid) {
+    res.status(400).send("Invalid id");
+    return;
+  }
+
+  const albums = await db
+    .knex("artist")
+    .join("album", "artist.id", "=", "album.artist_id")
+    .select(
+      "artist.id as artistId",
+      "artist.name as artist",
+      "artist.bio as bio",
+      "artist.artwork_url as artworkUrl",
+      "album.id as albumId",
+      "album.title as albumTitle",
+      "album.is_draft as isDraft",
+      "album.deleted as deleted"
+    )
+    .where("artist.id", "=", feedId)
+    .andWhere("album.deleted", false)
+    .andWhere("album.is_draft", false)
+    .catch((err) => {
+      log.debug(`Error querying albums table to generate artist feed: ${err}`);
+      res.status(404).send("No such feed");
+      return [];
+    });
+
+  if (albums.length > 0) {
+    const feed = await buildArtistFeed(albums);
+    res.send(feed);
+  } else {
+    res.status(404).send("No such feed");
```

Comment:
we should just use the else statement below, since its the same
```suggestion
      return [];
    });

  if (albums.length > 0) {
    const feed = await buildArtistFeed(albums);
    res.send(feed);
  } else {
    res.status(404).send("No such feed");
```